packages/block-library/src/post-featured-image/edit.js

```diff
@@ -124,11 +132,15 @@ function PostFeaturedImageDisplay( {
 			<InspectorControls>
 				<PanelBody title={ __( 'Link settings' ) }>
 					<ToggleControl
-						label={ sprintf(
-							// translators: %s: Name of the post type e.g: "post".
-							__( 'Link to %s' ),
-							postType
-						) }
+						label={
+							postType?.labels.singular_name
+								? sprintf(
+										// translators: %s: Name of the post type e.g: "post".
+										__( 'Link to %s' ),
+										postType.labels.singular_name.toLowerCase()
+								  )
+								: __( 'Link to post' )
```

Comment:
I was curious about whether we needed a default or not and it seems we don't based on [the docs](https://developer.wordpress.org/reference/functions/register_post_type/). So `labels` are filled.

>'labels'
(string[]) An array of labels for this post type. If not set, post labels are inherited for non-hierarchical types and page labels for hierarchical ones. See [get_post_type_labels()](https://developer.wordpress.org/reference/functions/get_post_type_labels/) for a full list of supported labels.

Reply:
We need a fallback for cases when the `postTypeSlug` context isn't available in Site Editor - https://github.com/WordPress/gutenberg/pull/39209#issuecomment-1071927282.